tests/integration/test_movielens.py

```diff
@@ -68,49 +67,51 @@ def test_load_pandas_df(
     title_example,
     genres_example,
     year_example,
+    tmpdir,
 ):
     """Test MovieLens dataset load into pd.DataFrame
     """
     # Test if correct data are loaded and local_cache_path works
-    with TemporaryDirectory() as tmp_dir:
-        # Test if can handle different size of header columns
-        header = ["a"]
-        df = load_pandas_df(size=size, local_cache_path=tmp_dir, header=header)
-        assert len(df) == num_samples
-        assert len(df.columns) == max(
-            len(header), 2
-        )  # Should load at least 2 columns, user and item
+    tmp_dir = str(tmpdir.mkdir("load_pandas_df"))

```

Comment:
It is worth considering a `try ... finally ...` or `with open ... as` for testing with a temp directory. The reason for this is that if the testing fails somewhere in the middle (after the temporary directory is created), the temporary directory will not be removed.

Reply:
I think this is the pytest fixture `tmpdir`, it is supposed to be erased after the test is finished. It would be good to make sure that this is the behavior

Reply:
Found interesting fact from the pytest page:

>Temporary directories are by default created as sub-directories of the system temporary directory. The base name will be pytest-NUM where NUM will be incremented with each test run. Moreover, **entries older than 3 temporary directories will be removed**.

So, you don't need to remove manually, but worth to note that it won't be removed right away.

Also, tmp-dir is unique for each test. E.g. `test_load_item_df(tmpdir)`,
tmpdir is `.... Temp\pytest-of-jumin\pytest-1\test_load_item_df_100k_1682_1_0/`

As you can see,  it creates a folder "pytest-of-USER' and create subfolder `pytest-1` (2, 3, ,...) where the last three of them will be kept, and create the children folders under the subfolder for each test function for each parameter. In the example, the test function is `test_load_item_df` and the params are `(100k, 1682, 1, 0)`

Reply:
>  entries older than 3 temporary directories will be removed.

let's say we download movielens in 4 of the tests within the same file, then is the first directory going to disappear? or is it 4 directories in different files?

Reply:
As long as the tests are under the same pytest run, it should be there. 
So, for your case, you will have one folder e.g. `pytest-n` and have 4 test folders under it.
the folder `pytest-n` will be removed after 2 additional pytest runs.